sklearn/impute.py

```diff
@@ -908,21 +933,21 @@ def transform(self, X):
 
         X, X_filled, mask_missing_values = self._initial_imputation(X)
 
-        # edge case: in case the user specifies 0 for n_imputations,
-        # then there is no need to do burn in and the result should be
-        # just the initial imputation (before clipping)
-        if self.n_imputations < 1:
+        # edge case: in case the user specifies 0 for n_burn_in,
+        # then there is no need to do further imputation and the result should
+        # be just the initial imputation (before clipping)
+        if self.n_iter < 1:
             return X_filled
 
-        X_filled = np.clip(X_filled, self._min_value, self._max_value)
+        # clip only the initial filledin values
```

Comment:
I don't really get why we would expect the initial imputation to require clipping.

Reply:
Huh, that's a good point. What I should probably do is this:

```
    if self.n_iter < 1:
        X_filled[mask_missing_values] = np.clip(X_filled[mask_missing_values],
                                                self._min_value,
                                                self._max_value)
        return X_filled
```

Reply:
Or actually just not clip before imputing.